### PR TITLE
Use integrator interface, add flame script

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR adds a flame script to the repo, and changes the monitored drivers to use OrdinaryDiffEq.jl's integrator interface. This should help make the analysis a bit more streamlined.

### CODE REVIEW

**Comment by simonbyrne** [comment]
Is the pprof-generated site static? Could we generate it on buildkite, and post it to a webserver?

**Comment by charleskawczynski** [comment]
> Is the pprof-generated site static? Could we generate it on buildkite, and post it to a webserver?

I suspect it is, and that sounds great, but I'm not sure how to do that. Could we try to add that later?

**Comment by simonbyrne** [comment]
Hmm, doesn't look like it is officially supported: https://github.com/google/pprof/issues/670

Though apparently just saving the html mostly works: https://github.com/google/pprof/issues/401#issuecomment-739576424

File: examples/hybrid/bubble_3d.jl [hunk 1]
```diff
@@ -317,6 +312,12 @@ sol = @timev solve(
     progress_message = (dt, u, p, t) -> t,
 );
 
+if haskey(ENV, "CI_PERF_SKIP_RUN") # for performance analysis
+    throw(:exit_profile)
```
Comment (simonbyrne) [inline examples/hybrid/bubble_3d.jl:316]: should this just be `exit()`?

Comment (simonbyrne) [reply examples/hybrid/bubble_3d.jl:316]: oh, you catch it in the other script: in that case, maybe put the rest in a block?

Comment (charleskawczynski) [reply examples/hybrid/bubble_3d.jl:316]: I'm not sure I follow. FWIW, these lines were only moved

**Review by simonbyrne — APPROVED** [review]

**Comment by jakebolewski** [comment]
In addition to or instead of PProf could we generate a cpuprofile file assest? 

https://github.com/julia-vscode/ChromeProfileFormat.jl

You can view these directly in vscode after downloading the file.

**Comment by charleskawczynski** [comment]
> In addition to or instead of PProf could we generate a cpuprofile file assest?

Opened #483.

**Pull request merged**